### PR TITLE
stagingapi: submit_to_prj(): create sub-packages for adi requests.

### LINKED ISSUE #946: adi command should try to create sub-package in case multiple-specfile
adi command should to create sub-package in case multiple-specfile, otherwise sub-package potentially build failed after merged into Factory as it is never built in adi staging.

### PR DESCRIPTION
**Pull request by jberry-suse**
Rather non-trivial to test given testing setup. If anyone has a case to try it on that would be great.

Fixes #946.

### CODE REVIEW

File: osclib/stagingapi.py [hunk 1]
```diff
@@ -807,8 +807,10 @@ def rm_from_prj(self, project, package=None, request_id=None,
 
         for sub_prj, sub_pkg in self.get_sub_packages(package):
             sub_prj = self.map_ring_package_to_subject(project, sub_pkg)
-            if sub_prj != subprj:  # if different to the main package's prj
-                delete_package(self.apiurl, sub_prj, sub_pkg, force=True, msg=msg)
+            # Skip inner-project links for letter stagings as they should
+            # already be in place, but for adi packages remove them.
+            if not self.is_adi_project(project) and sub_prj == project: continue
```
Comment (nilxam) [inline osclib/stagingapi.py]: I think it should be ``sub_prj == subprj`` due to ``project`` does not go through ``map_ring_package_to_subject()``

**Comment by nilxam** [comment]
I'll test it on https://build.opensuse.org/project/show/openSUSE:Factory:Staging:adi:8

File: osclib/stagingapi.py (hunk 1 above)
Comment (jberry-suse) [inline osclib/stagingapi.py]: I would expect `map_ring_package_to_subject()` to have no effect on an adi project given:
```python
if self.is_adi_project(project):
    return project
```
As such it should not matter, but seems better stylistically to check project.

**Comment by nilxam** [comment]
I did a small test, this does not work, because https://github.com/openSUSE/osc-plugin-factory/blob/master/osclib/stagingapi.py#L1073 returned the empty list as the request went to adi means it doesn't exist in ring, we might need to check ``.spec`` from souce in this case.

**Comment by jberry-suse** [comment]
Can that method just be adapted to look at the target package in the main project and fallback to source project for new packages?

**Comment by jberry-suse** [comment]
This can likely be fixed by making a call against the source project using the same call the old repo checker used.

**Comment by jberry-suse** [comment]
Closing in favor of #1110.

**Pull request closed without merging**